packages/compass-smoke-tests/src/dispatch.ts

```diff
@@ -0,0 +1,188 @@
+import assert from 'node:assert/strict';
+import cp from 'node:child_process';
+
+import * as github from '@actions/github';
+import createDebug from 'debug';
+
+const GITHUB_OWNER = 'mongodb-js';
+const GITHUB_REPO = 'compass';
+const GITHUB_WORKFLOW_ID = 'test-installers.yml';
+const MAX_GET_LATEST_ATTEMPTS = 10;
+const WATCH_POLL_TIMEOUT_MS = 1000 * 60 * 60; // 1 hour
+
+const debug = createDebug('compass:smoketests:dispatch');
+
+async function getLatestRun(octokit: ReturnType<typeof github.getOctokit>) {
+  const {
+    data: {
+      total_count: totalCount,
+      workflow_runs: [latestRun],
+    },
+  } = await octokit.rest.actions.listWorkflowRuns({
+    owner: GITHUB_OWNER,
+    repo: GITHUB_REPO,
+    workflow_id: GITHUB_WORKFLOW_ID,
+  });
+  assert(
+    totalCount > 0 && typeof latestRun === 'object',
+    'Expected at least one prior run'
+  );
+  debug('Found latest run', {
+    runNumber: latestRun.run_number,
+    url: latestRun.html_url,
+  });
+  return latestRun;
+}
+
+type GetNextRunOptions = {
+  octokit: ReturnType<typeof github.getOctokit>;
+  expectedRunNumber: number;
+
+  /**
+   * Delay in milliseconds to wait between requests when polling for the next run.
+   */
+  pollDelayMs: number | undefined;
+
+  /**
+   * Delay in milliseconds to wait before issuing a request to check that no other runs has been started.
+   */
+  sanityDelayMs: number | undefined;
+};
+
+async function getNextRun({
+  octokit,
+  expectedRunNumber,
+  pollDelayMs,
+  sanityDelayMs,
+}: GetNextRunOptions) {
+  for (let attempt = 0; attempt < MAX_GET_LATEST_ATTEMPTS; attempt++) {
+    debug('Getting latest run', { attempt, expectedRunNumber });
+    const run = await getLatestRun(octokit);
+    if (run.run_number === expectedRunNumber) {
+      debug(
+        'Found the expected run, waiting before requesting again to check sanity',
+        {
+          sanityDelayMs,
+          runNumber: run.run_number,
+          url: run.html_url,
+        }
+      );
+      assert.equal(
+        run.event,
+        'workflow_dispatch',
+        'Expected the new run to be a dispatch'
+      );
+
+      await new Promise((resolve) => setTimeout(resolve, sanityDelayMs));
+      debug('Getting latest run again - expecting it to be the same');
+      const latestRun = await getLatestRun(octokit);
+      assert.equal(
+        latestRun.run_number,
+        expectedRunNumber,
+        'Runs are being dispatched too quickly, for the heuristics to be accurate.'
+      );
```

Comment:
Doing a sanity check by waiting some time (default is 10s) from the workflow run has been identified to re-fetch and assert this is still the latest run. This will make the command fail if there's too many runs being dispatched for the heuristics (i.e. the latest run is the one we just dispatched) to be accurate.